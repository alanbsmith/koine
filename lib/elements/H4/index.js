import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import { rem } from 'polished';

import { fontStyles, fontWeights } from '../../modifiers';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};

const H4 = styled.h4`
  font-size: ${rem(28)};
  margin: 0;
  text-transform: capitalize;
  ${applyStyleModifiers(modifierConfig)}
`;

export default H4;
