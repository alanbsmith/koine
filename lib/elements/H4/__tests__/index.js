import React from 'react';
import H4 from '../index';
import renderer from 'react-test-renderer';

import { renderWithTheme } from '__tests__/helpers';


describe('H4 Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <H4>Header Element</H4>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies normalWeight modifier', () => {
    const tree = renderWithTheme(
      <H4 modifiers={['normalWeight']}>Header Element</H4>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
