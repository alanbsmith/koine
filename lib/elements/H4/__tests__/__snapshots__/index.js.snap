// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H4 Element applies normalWeight modifier 1`] = `
<h1
  className="sc-bdVaJa bMPpFu"
>
  Header Element
</h1>
`;

exports[`H4 Element renders correctly 1`] = `
<h1
  className="sc-bdVaJa fzsVZC"
>
  Header Element
</h1>
`;
